infrastructure-formula/infrastructure/libvirt/domains.sls

```diff
@@ -67,6 +68,55 @@ vm_uuid_map_{{ machine }}:
   - name: /etc/uuidmap
   - text: '{{ machine }}: {{ salt['cmd.run']('uuidgen') }}'
   - unless: 'grep -q {{ machine }} /etc/uuidmap'
+
+{%- if clusterpillar[cluster].get('storage') == 'local' and 'image' in config %}
+{%- if 'root' in config['disks'] %}
+{%- set root_disk = topdir ~ '/disks/' ~ machine ~ '_root.qcow2' %}
+{%- set image = topdir ~ '/os-images/' ~ config['image'] %}
+write_vmdisk_{{ machine }}_root:
+  file.copy:
+    - name: {{ root_disk }}
+    - source: {{ image }}
+    {%- if not pillar.get('force_vmdisk', False) %}
+    - unless: test -f {{ root_disk }}
+    {%- endif %}
+
+{%- set disk_size = config['disks']['root'] %}
+{%- set image_info = salt['cmd.run']('qemu-img info --out json ' ~ image) | load_json %}
+{%- set image_size = image_info['virtual-size'] %}
+
+{%- if disk_size.endswith('G') %}
+{%- set converted_size = ( disk_size.rstrip('G') | int * 1073741824 ) | int %}
+{%- else %}
+{%- do salt.log.error('infrastructure.libvirt: illegal disk size ' ~ disk_size ~ ' for machine ' ~ machine) %}
```

Comment:
A hint like `not ending with 'G'` in the error message might be helpful ;-)

Reply:
Ok!